webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java

```diff
@@ -345,8 +360,112 @@ public String saveSiteImage(HttpServletRequest req, @PathVariable String siteId)
             siteService.save(site);
             return edit.getUrl();
         } catch (Exception e) {
-            log.error("Failed to update image for site {}", siteId, e);
+            log.error("Failed to update image for site {}: {}", siteId, e.toString());
             throw e;
         }
-	}
+    }
+
+    @Override
+    public String getLabel() {
+        return "dashboard";
+    }
+
+    @Override
+    public boolean parseEntityReference(String reference, Reference ref) {
+
+        if (!reference.startsWith(REFERENCE_ROOT)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public Optional<String> getTool() {
+        return Optional.of(DASHBOARD_TOOL_ID);
+    }
+
+    @Override
+    public String[] myToolIds() {
+        return new String[] { DASHBOARD_TOOL_ID };
+    }
+
+    @Override
+    public Map<String, String> transferCopyEntities(String fromContext, String toContext, List<String> ids, List<String> options) {
+
+        try {
+            Site fromSite = siteService.getSite(fromContext);
+            String fromConfig = fromSite.getProperties().getProperty("dashboard-config");
+            if (fromConfig != null) {
+                Site toSite = siteService.getSite(toContext);
+                toSite.getProperties().addProperty("dashboard-config", fromConfig);
+                siteService.save(toSite);
+            }
+        } catch (IdUnusedException idue) {
+            log.error("No site found for {} or {}", fromContext, toContext);
+        } catch (Exception e) {
+            log.error("Failed to copy the dashboard config: {}", e.toString());
+        }
+
+        Map<String, String> map = Collections.EMPTY_MAP;
+
+        String fromCollectionId = contentHostingService.getSiteCollection(fromContext);
+
+        if (fromCollectionId == null) return map;
+
+        try {
+            contentHostingService.checkCollection(fromCollectionId);
+        } catch (Exception e) {
+            log.warn("No access to site {}'s content collection", fromContext);
+            return map;
+        }
+
+        String toCollectionId = contentHostingService.getSiteCollection(toContext);
+
+        try {
+            contentHostingService.checkCollection(toCollectionId);
+        } catch (Exception e) {
+            try {
+                contentHostingService.commitCollection(contentHostingService.addCollection(toCollectionId));
+            } catch (Exception e2) {
+                log.error("Failed to add collection {}: {}", toCollectionId, e2.toString());
+            }
+        }
+
+        String sourceId = fromCollectionId + "course_image.png";
```

Comment:
Declare "course_image.png" as a static variable at the top of the method and change also here:
https://github.com/sakaiproject/sakai/pull/12415/files#diff-d82b0ddd3190c49087dafe5ae0c49f8f0888bd270ebaf61d98e6a7af6db55019R444